### PR TITLE
Dump Behat page data on failed step

### LINKED ISSUE #1022: Behat - dump page source if test has failed
Important to quickly identify failed tests like this one: https://travis-ci.org/github/atk4/ui/jobs/668304912#L1017

### PR DESCRIPTION
**Pull request by mvorisek**
Fixes #1022

### CODE REVIEW

**Comment by mvorisek** [comment]
PR complete, more than needed as the current `develop` tests are broken (issue #1043).

Is it ok to add `j0k3r/php-imgur-api-client` to `dev` composer dependencies only for this PR? Or do you have a better idea where to upload a creen of a failed test?

Screenshots are important for quick debugging and even more important for AJAX tests, where the main page source may be not enought.

**Comment by DarkSide666** [comment]
I think it's OK to use `j0k3r/php-imgur-api-client` in `require-dev` section if that helps us to identify issues.

**Comment by romaninsh** [comment]
I wonder if there is a way to attach screenshot to the issue through a comment.

**Comment by romaninsh** [comment]
On the PR - it only adds the method. At least some documentation is necessary on how others can use it.

File: features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -26,6 +28,30 @@ public function getSession($name = null)
         return $this->getMink()->getSession($name);
     }
 
+    /**
+     * Dump current page data when step failed to allow easy debug on TravisCI.
+     *
+     * @AfterStep
+     */
+    public function dumpPageAfterFailedStep(AfterStepScope $event)
```
Comment (romaninsh) [inline features/bootstrap/FeatureContext.php:37]: do we need a place to store screenshot? I can provision a public S3 bucket for that.

**Review by romaninsh — CHANGES_REQUESTED** [review]

**Comment by romaninsh** [comment]
I can provide an S3 bucket where image(s) could be uploaded.

File: features/bootstrap/FeatureContext.php (hunk 1 above)
Comment (mvorisek) [reply features/bootstrap/FeatureContext.php:37]: > 
> 
> On the PR - it only adds the method. At least some documentation is necessary on how others can use it.

It has on step hook, so it is added automatically activated for every failed step.

Comment (mvorisek) [reply features/bootstrap/FeatureContext.php:37]: > 
> 
> do we need a place to store screenshot? I can provision a public S3 bucket for that.

I was planning to use imgur.com, but S3 is good - can you please add it directly to this PR? And please - do not forget to handle automatic expires after some time + always put there a link to delete - important, when some sensitive data will be dumpled/saved by a mistake.

S3 can be also used for dumping the page source (currently printed to stdout in this PR).

**Comment by mvorisek** [comment]
replaced with https://github.com/atk4/ui/pull/1281

**Pull request closed without merging**